Library/Homebrew/brew.sh

```diff
@@ -744,6 +744,30 @@ fi
 source "${HOMEBREW_LIBRARY}/Homebrew/utils/analytics.sh"
 setup-analytics
 
+# Migrate from linuxbrew-core to homebrew-core
+# HOMEBREW_CORE_MERGE_MAINTAINER can be set by the maintainers
+# who merge homebrew-core into linuxbrew-core to avoid being migrated
+# We only migrate HOMEBREW_DEVELOPER users right now
+# Don't migrate on CI either
+if [[ -n "${HOMEBREW_DEVELOPER}" ]] &&
+   [[ -z "${HOMEBREW_CORE_MERGE_MAINTAINER}" ]] &&
+   [[ -z "${GITHUB_ACTIONS}" ]] &&
+   [[ "${HOMEBREW_BASH_COMMAND}" == "${HOMEBREW_LIBRARY}/Homebrew/cmd/update.sh" ]] &&
+   [[ "https://github.com/Homebrew/linuxbrew-core" == "$(git -C "${HOMEBREW_CORE_REPOSITORY}" remote get-url origin)" ]]
+then
+  export HOMEBREW_CORE_DEFAULT_GIT_REMOTE="https://github.com/Homebrew/homebrew-core"
+  export HOMEBREW_CORE_GIT_REMOTE="${HOMEBREW_CORE_DEFAULT_GIT_REMOTE}"
+
+  if [[ "$(git config homebrew.linuxbrewmigrationcomplete 2>/dev/null)" != "true" ]]
+  then
+    export HOMEBREW_LINUXBREW_MIGRATION=1
```

Comment:
This is a change. Now, `HOMEBREW_LINUXBREW_MIGRATION` is only set if we haven't completed the entire migration. However, regardless of whether this is set or not, the remote will still be updated.

So, for example, if the migration is fully completed but then for some reason the remote gets switched back, it will still be fixed on the next `brew.sh` invocation. However, `HOMEBREW_LINUXBREW_MIGRATION` will not be set so the `version_scheme`s will not be changed back to `-1` (assuming it was a `brew update` call).